Test Plans General Refactor (#779)
italia/eidas-it-wallet-docs: docs/en/test-plans-wallet.rst

    - Device Attestation is successful and results are included in the Wallet Attestation payload.
  * - WS-022
    - Attestation
    - The Wallet Attestation MUST include a signature using the Wallet Binding Key, and the certificate chain MUST be verifiable to a trusted root.
    - Signature is present, valid, and verifiable using the provided certificate chain.
  * - WS-023
    - Attestation
    - The Wallet MUST include a Device Attestation result in the Wallet Attestation structure.
    - A valid Device Attestation object (Play Integrity or DC App Attest result) is embedded in the Attestation.
  * - WS-024
    - Backup & Restore
    - During Restore, the Wallet MUST validate the integrity of the encrypted backup file using an integrity check mechanism.
    - The Wallet refuses to restore a tampered or corrupted backup file.
  * - WS-025
    - Revocation
    - In case of Wallet Instance Revocation, the Wallet MUST delete any locally stored Verifiable Credentials.
    - No credential data remains accessible after revocation is triggered.
  * - WS-026
    - Revocation

>     - The Wallet MUST notify the Wallet Backend with a Revocation Request that includes a valid proof of possession of the Wallet Binding Key.

Reviewer: ```suggestion
    - The Wallet MUST notify the Wallet Backend with a Revocation Request containing a 'status' parameter set to 'REVOKED'.
```

    - Wallet Attestation includes a valid DC App Attest JWT response signed by Apple.
  * - WS-033
    - Security
    - The Wallet MUST verify that the Play Integrity token signature is valid and issued by Google.
    - The Wallet rejects invalid or forged Play Integrity tokens.
  * - WS-034
    - Security
    - The Wallet MUST validate that the 'nonce' value used in Play Integrity is cryptographically bound to the Wallet Instance.
    - Any tampering with the nonce is detected and leads to Attestation rejection.
  * - WS-035
    - Attestation
    - The Wallet MUST send the Wallet Attestation to the Wallet Backend during registration.
    - Wallet Backend receives the attestation and verifies its validity.
  * - WS-036
    - Backup & Restore
    - The Wallet MUST encrypt backups using a symmetric key derived from User secrets.
    - The backup cannot be decrypted without the original User authentication material.
  * - WS-037
    - Backup & Restore
    - The Wallet MUST include metadata in the backup that identifies the version and creation timestamp.
    - Restore process reads and verifies backup metadata before proceeding.
  * - WS-038
    - Revocation

>     - The Wallet MUST send a signed Revocation Request including the Wallet Binding Key signature to the Backend.

Reviewer: ```suggestion
    - The Wallet MUST send a Revocation Request to the Backend endpoint, including the Wallet ID as a path parameter.
 ```
